src/fast_forward.rs

```diff
@@ -79,6 +80,33 @@ fn curry_and_treehash(inner_puzzle_hash: &Bytes32, singleton_struct: &SingletonS
     .into()
 }
 
+pub fn update_eligible_for_ff(a: &Allocator, spend: &mut Spend) {
+    if (spend.flags & ELIGIBLE_FOR_FF) == 0 {
+        return;
+    }
+
+    let eligible = (|| {
+        // a spend eligible for fast-forward must be a singleton, which use odd amounts
+        if (spend.coin_amount & 1) != 1 {
+            return false;
+        }
+
+        // if this still looks like it might be a singleton, check the output coins
+        // to look for something that looks like a singleton output, with the same
+        // puzzle hash as our input coin
+        // if there is at least one output coin that has the same puzzle hash
+        // and an odd coin amount, it looks like a valid singleton
+        spend
+            .create_coin
+            .iter()
+            .any(|c| (c.amount & 1) == 1 && a.atom(spend.puzzle_hash) == c.puzzle_hash)
```

Comment:
I made a logic change here as well, to not require the amounts to be the same, just for the new amount to also be odd (as that's a requirement for singletons)